agent/config/config.go

```diff
@@ -237,6 +241,25 @@ func get(args []string, cfg *Config, l *logrus.Entry) (configFileF string, err e
 			cfg.Paths.ExportersBase = abs
 		}
 
+		if cfg.Paths.TempDir == "" {
+			wd, err := os.Getwd()
+			if err == nil {
+				cfg.Paths.TempDir = filepath.Join(wd, agentTmpPath)
+			} else {
+				// As a last resort, if unable to get the workdir, use system temp directory.
+				cfg.Paths.TempDir = filepath.Join(os.TempDir(), agentTmpPath)
```

Comment:
I think we need to warn about that.

Reply:
What if we do not try to put the config file to `os.TempDir()` since this is essentially where the issue originated from?

Reply:
that sounds good to me

Reply:
Done.